pkg/scheduler/framework/plugins/noderesources/topology_match.go

```diff
@@ -0,0 +1,316 @@
+}
+
+type NUMANodeList []NUMANode
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (tm *TopologyMatch) Name() string {
+	return TopologyMatchName
+}
+
+func filter(containers []v1.Container, nodes NUMANodeList, qos v1.PodQOSClass) *framework.Status {
+	if qos == v1.PodQOSBestEffort {
+		return nil
+	}
+
+	zeroQuantity := resource.MustParse("0")
+	for _, container := range containers  {
+		bitmask := bm.NewEmptyBitMask()
+		bitmask.Fill()
+		for resource, quantity := range container.Resources.Requests {
+			resourceBitmask := bm.NewEmptyBitMask()
+			for _, numaNode := range nodes {
+				numaQuantity, ok := numaNode.Resources[resource]
+				// if can't find requested resource on the node - skip (don't set it as available NUMA node)
+				// if unfound resource has 0 quantity probably this numa node can be considered
+				if !ok && quantity.Cmp(zeroQuantity) != 0{
+					continue
+				}
+				// Check for the following:
+				// 1. set numa node as possible node if resource is memory or Hugepages (until memory manager will not be merged and
+				// memory will not be provided in CRD
+				// 2. set numa node as possible node if resource is cpu and it's not guaranteed QoS, since cpu will flow
+				// 3. set numa node as possible node if zero quantity for non existing resource was requested (TODO check topology manaager behaviour)
+				// 4. otherwise check amount of resources
+				if resource == v1.ResourceMemory ||
+					strings.HasPrefix(string(resource), string(v1.ResourceHugePagesPrefix)) ||
+					resource == v1.ResourceCPU && qos != v1.PodQOSGuaranteed ||
+					quantity.Cmp(zeroQuantity) == 0 ||
+					numaQuantity.Cmp(quantity) >= 0 {
+					resourceBitmask.Add(numaNode.NUMAID)
+				}
+			}
+			bitmask.And(resourceBitmask)
+		}
+		if bitmask.IsEmpty() {
+			// definitly we can't align container, so we can't align a pod
+			return framework.NewStatus(framework.Unschedulable, fmt.Sprintf("Can't align container: %s", container.Name))
+		}
+	}
+	return nil
+}
+
+// checkTopologyPolicies return true if we're working with such policy
+func getTopologyPolicies(nodeTopologies nodeTopologyMap, nodeName string) []v1.TopologyManagerPolicy {
+	if nodeTopology, ok := nodeTopologies[nodeName]; ok {
+		policies := make([]v1.TopologyManagerPolicy, 0)
+		for _, policy := range nodeTopology.TopologyPolicies {
+			policies = append(policies, v1.TopologyManagerPolicy(policy))
+		}
+		return policies
+	}
+	return nil
+}
+func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
+	res := make(v1.ResourceList)
+	for resName, resInfo := range zone.Resources {
+		quantity, err := resource.ParseQuantity(resInfo.Allocatable)
+		if err != nil {
+			klog.Errorf("Failed to parse %s", resInfo.Allocatable)
+			continue
+		}
+		res[resName] = quantity
+	}
+	return res
+}
+
+func (tm TopologyMatch) PolicyFilter(pod *v1.Pod, zoneMap topologyv1alpha1.ZoneMap) *framework.Status {
+	if v1qos.GetPodQOS(pod) != v1.PodQOSGuaranteed {
```

Comment:
@AlexeyPerevalov Was testing out this latest implementation and was thinking about pod with resources as can be seen below where resource request= limit. In such cases, v1qos.GetPodQOS returns Burstable QoS (as memory is missing I guess?). 
```
resources:
      limits:
        cpu: 1
        example.com/deviceA: 1
        example.com/deviceB: 1
      requests:
        cpu: 1
        example.com/deviceA: 1
        example.com/deviceB: 1
```
This check here completely disregards such pods as they are not in the guaranteed QoS class, we might want to address this at least for the time being where there is no way to align memory based on topology?

Reply:
Memory in requests is necessary for guaranteed QoS. But in topology manager makes alignment for all QoS, and such behaviour was here too. Looks like it was missed during last rebase/factoring out. I'll fix it.